pkg/secrets/secrets.go

```diff
@@ -17,7 +33,56 @@ func ValidateCredentials(secret *v1.Secret) error {
 		return ValidateAWSCredentials(secret)
 	case AzureSecretType:
 		return ValidateAzureCredentials(secret)
+	case GCPSecretType:
+		return ValidateGCPCredentials(secret)
+	case FilestoreSecretType:
+		return ValidateFileStoreCredentials(secret)
 	default:
 		return errors.Errorf("Unsupported type '%s' for secret '%s:%s'", string(secret.Type), secret.Namespace, secret.Name)
 	}
 }
+
+func getLocationType(secret *corev1.Secret) (repositoryserver.RepositoryServerSecret, error) {
+	var locationType []byte
+	var ok bool
+	if secret == nil {
+		return nil, errors.New("Secret is Nil")
+	}
+
+	if locationType, ok = (secret.Data[repositoryserver.TypeKey]); !ok {
+		return nil, errors.Errorf("secret '%s:%s' does not have required field %s", secret.Namespace, secret.Name, repositoryserver.TypeKey)
+	}
+
+	switch repositoryserver.LocType(string(locationType)) {
+	case repositoryserver.LocTypeS3:
+		return repositoryserver.NewAWSLocation(secret), nil
+	case repositoryserver.LocTypeAzure:
+		return repositoryserver.NewAzureLocation(secret), nil
+	case repositoryserver.LocTypeGCS:
+		return repositoryserver.NewGCPLocation(secret), nil
+	case repositoryserver.LocTypeFilestore:
+		return repositoryserver.NewFileStoreLocation(secret), nil
+	default:
+		return nil, errors.Errorf("Unsupported location type '%s' for secret '%s:%s'", locationType, secret.Namespace, secret.Name)
+	}
+}
+
+func ValidateRepositoryServerSecret(repositoryServerSecret *corev1.Secret) error {
+	var secret repositoryserver.RepositoryServerSecret
+	var err error
+
+	switch repositoryServerSecret.Type {
+	case repositoryserver.Location:
+		secret, err = getLocationType(repositoryServerSecret)
```

Comment:
The function name doesn't convey its purpose. Consider renaming it to something meaningful. Maybe `getLocationSecret()`, or better